Block/Service/AbstractCategoriesBlockService.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\ClassificationBundle\Block\Service;
+
+use Sonata\AdminBundle\Admin\AdminInterface;
+use Sonata\AdminBundle\Form\FormMapper;
+use Sonata\BlockBundle\Block\BlockContextInterface;
+use Sonata\BlockBundle\Model\BlockInterface;
+use Sonata\ClassificationBundle\Model\CategoryInterface;
+use Sonata\ClassificationBundle\Model\CategoryManagerInterface;
+use Sonata\ClassificationBundle\Model\ContextManagerInterface;
+use Sonata\CoreBundle\Model\Metadata;
+use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Christian Gripp <mail@core23.de>
+ */
+abstract class AbstractCategoriesBlockService extends AbstractClassificationBlockService
+{
+    /**
+     * @var CategoryManagerInterface
+     */
+    private $categoryManager;
+
+    /**
+     * @var AdminInterface
+     */
+    private $categoryAdmin;
+
+    /**
+     * @param string                   $name
+     * @param EngineInterface          $templating
+     * @param ContextManagerInterface  $contextManager
+     * @param CategoryManagerInterface $categoryManager
+     * @param AdminInterface           $categoryAdmin
+     */
+    public function __construct($name, EngineInterface $templating, ContextManagerInterface $contextManager, CategoryManagerInterface $categoryManager, AdminInterface $categoryAdmin)
```

Comment:
I don't think we should discuss it here (even if I tend to agree with @greg0ire ). Also, communication between to parties can hardly be called a "discussion". But if the decision is to follow symfony instead of following logic (whatever logic it might be), then I don't think we should start convincing anyone to do otherwise.